frameioclient/download.py

```diff
@@ -1,18 +1,45 @@
-import requests
-import math
 import os
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
 
 class FrameioDownloader(object):
-  def __init__(self, asset, download_folder):
+  def __init__(self, asset, download_folder, replace):
     self.asset = asset
     self.download_folder = download_folder
+    self.replace = replace
+    self.retry_strategy = Retry(
+        total=3,
+        backoff_factor=1,
+        status_forcelist=[429],
```

Comment:
For this, I think you'll want to add more statuses so that it catches on more than just a rate limit. I also think that there might be other ways of ensuring that we retry a failed download.

Reply:
Ok! Might have been a little quick here. Testing some other error codes. 

Do you think we should add a xxhash check here and retry the download if it fails? Because then I can work on a solution to retry 3x times for example if hashes don't match.

Reply:
That's not a bad idea! If you want to add it, feel free.

Reply:
Doing the xxhash on download is much easier than on upload, because of the delay on the upload side that would require putting the QC into a queue and then managing that state and creating a background thread to check on it continuously.

Reply:
Check the latest commit 😉 and yes uploads are a bit trickier!